src/client/dfs/dfs.c

```diff
@@ -1510,12 +1751,12 @@ open_sb(daos_handle_t coh, bool create, daos_obj_id_t super_oid,
 	D_DEBUG(DB_ALL, "DFS Library Layout version: %d\n", DFS_LAYOUT_VERSION);
 
 	*ver = layout_ver;
-	attr->da_chunk_size = (chunk_size) ? chunk_size :
-		DFS_DEFAULT_CHUNK_SIZE;
+	attr->da_chunk_size = (chunk_size) ? chunk_size : DFS_DEFAULT_CHUNK_SIZE;
 	attr->da_oclass_id = oclass;
-
-	/** DFS_RELAXED by default */
+	attr->da_dir_oclass_id = dir_oclass;
 	attr->da_mode = mode;
+	if (iods[HINTS_IDX].iod_size != 0)
+		strcpy(attr->da_hints, hints);
```

Comment:
strncpy?

Reply:
this hint string is coming from DAOS not the user, so it's safe here too